### PR TITLE
[WIP] zemeroth: init at 0.6.0

### PR DESCRIPTION
**Pull request by 0x4A6F**
###### Motivation for this change

[0.6.0 released](https://ozkriff.games/2019-09-21--devlog-zemeroth-v0-6/)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by 0x4A6F** [comment]
- Running zemeroth leads to:
```
$ /nix/store/lq80zj4djfnrgc37h8g0sy9hwm48q3px-zemeroth-0.6.0/bin/zemeroth
Error: IOError(Os { code: 2, kind: NotFound, message: "No such file or directory" })
```

- With a clone of zemeroth_assets under ./assets in $PWD
```
$ /nix/store/lq80zj4djfnrgc37h8g0sy9hwm48q3px-zemeroth-0.6.0/bin/zemeroth
Error: GgezError(WindowCreationError(NotSupported("both libglx and libEGL not present")))
```

- Here is a strace:
```
openat(AT_FDCWD, "/nix/store/9ciymhmianf5jk8j2w81pgs5crlzxhs1-sane-config/lib/sane/libGLX.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/libGLX.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver-32/lib/libGLX.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/mzyl7nab4n66ad2dg3m9kvf9285mms1r-systemd-243-lib/lib/libGLX.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/8i03vc4n2a6lhd7s713ahhnhhay04j4a-alsa-lib-1.1.9/lib/libGLX.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libGLX.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libGLX.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/9ciymhmianf5jk8j2w81pgs5crlzxhs1-sane-config/lib/sane/libGLX.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/libGLX.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver-32/lib/libGLX.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/mzyl7nab4n66ad2dg3m9kvf9285mms1r-systemd-243-lib/lib/libGLX.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/8i03vc4n2a6lhd7s713ahhnhhay04j4a-alsa-lib-1.1.9/lib/libGLX.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libGLX.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libGLX.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/9ciymhmianf5jk8j2w81pgs5crlzxhs1-sane-config/lib/sane/libEGL.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/libEGL.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver-32/lib/libEGL.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/mzyl7nab4n66ad2dg3m9kvf9285mms1r-systemd-243-lib/lib/libEGL.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/8i03vc4n2a6lhd7s713ahhnhhay04j4a-alsa-lib-1.1.9/lib/libEGL.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libEGL.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libEGL.so.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/9ciymhmianf5jk8j2w81pgs5crlzxhs1-sane-config/lib/sane/libEGL.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/libEGL.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver-32/lib/libEGL.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/mzyl7nab4n66ad2dg3m9kvf9285mms1r-systemd-243-lib/lib/libEGL.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/8i03vc4n2a6lhd7s713ahhnhhay04j4a-alsa-lib-1.1.9/lib/libEGL.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libEGL.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/lib/libEGL.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

- current /run/opengl-driver/lib/:
```
total 25
lrwxrwxrwx 1 root root 71 Jan  1  1970 d3d -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/d3d
dr-xr-xr-x 2 root root 24 Jan  1  1970 dri
lrwxrwxrwx 1 root root 77 Jan  1  1970 intel-ocl -> /nix/store/8rfhh9p450448snyr6b8155pdql7jzsq-intel-ocl-5.0-63503/lib/intel-ocl
lrwxrwxrwx 1 root root 82 Jan  1  1970 libEGL_mesa.so -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libEGL_mesa.so
lrwxrwxrwx 1 root root 84 Jan  1  1970 libEGL_mesa.so.0 -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libEGL_mesa.so.0
lrwxrwxrwx 1 root root 88 Jan  1  1970 libEGL_mesa.so.0.0.0 -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libEGL_mesa.so.0.0.0
lrwxrwxrwx 1 root root 82 Jan  1  1970 libGLX_mesa.so -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libGLX_mesa.so
lrwxrwxrwx 1 root root 84 Jan  1  1970 libGLX_mesa.so.0 -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libGLX_mesa.so.0
lrwxrwxrwx 1 root root 88 Jan  1  1970 libGLX_mesa.so.0.0.0 -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libGLX_mesa.so.0.0.0
lrwxrwxrwx 1 root root 86 Jan  1  1970 libvulkan_intel.so -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libvulkan_intel.so
lrwxrwxrwx 1 root root 87 Jan  1  1970 libvulkan_radeon.so -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libvulkan_radeon.so
lrwxrwxrwx 1 root root 83 Jan  1  1970 libxatracker.so -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libxatracker.so
lrwxrwxrwx 1 root root 85 Jan  1  1970 libxatracker.so.2 -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libxatracker.so.2
lrwxrwxrwx 1 root root 89 Jan  1  1970 libxatracker.so.2.5.0 -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libxatracker.so.2.5.0
lrwxrwxrwx 1 root root 85 Jan  1  1970 libXvMCnouveau.so -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libXvMCnouveau.so
lrwxrwxrwx 1 root root 82 Jan  1  1970 libXvMCr600.so -> /nix/store/7f8acg7iqsii9n5bs21ar11pvmg430d4-mesa-19.1.5-drivers/lib/libXvMCr600.so
dr-xr-xr-x 2 root root 20 Jan  1  1970 vdpau
```

Open:
- [ ] Use assets from store path
- [ ] Working libGLX
- [ ] Working libEGL

**Comment by Lassulus** [comment]
@GrahamcOfBorg eval

**Comment by Lassulus** [comment]
hmm, for some reason ci can't import libudev

File: pkgs/games/zemeroth/default.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+, runCommand
+, rustPlatform
+, pkgconfig
+, makeWrapper
+, makeDesktopItem
+, alsaLib
+, libGL
+, libpthreadstubs
+, libSM
+, libudev
```
Comment (Lassulus) [inline pkgs/games/zemeroth/default.nix]: maybe we can change this to just udev, libudev is an alias to udev since 2018-04-25

Comment (0x4A6F) [reply pkgs/games/zemeroth/default.nix]: Changed to udev.

**Comment by Lassulus** [comment]
sup?

**Pull request closed without merging**